src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public Node<K, V> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<K, V> next) {
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
 
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if ((key == null && current.getKey() == null)
+                        || (key != null && key.equals(current.getKey()))) {
+                    current.setValue(value);
+                    return;
+                }
+                if (current.getNext() == null) {
+                    current.setNext(newNode);
+                    break;
+                }
+                current = current.getNext();
+            }
+        }
+
+        size++;
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> current = table[index];
+
+        while (current != null) {
+            if ((key == null && current.getKey() == null)
+                    || (key != null && key.equals(current.getKey()))) {
+                return current.getValue();
+            }
+            current = current.getNext();
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        threshold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
+
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                int index = (node.getKey() == null) ? 0
+                        : Math.abs(node.getKey().hashCode() % newCapacity);
```

Comment:
In the `resize` method, the calculation of the new index should ensure that the index is non-negative. Similar to the `getIndex` method, consider using `(node.getKey().hashCode() & 0x7FFFFFFF) % newCapacity` to avoid negative indices.